### DISCUSSION TITLE
Smart component with remote submit

### DISCUSSION THREAD

**Discussion by qerijo** [react-hook-form/react-hook-form#6589] [Q&A] [answered]
Hi. I have a modal with a form and buttons outside of it. If I want to create a smart form component with an ability to submit it remotely, is it okay to just pass useHook methods into it rather then calling useHook inside as in the example or there could be a better way to approach it?

**Comment by bluebill1049** [comment, marked as answer]
`handleSubmit()()`?

**Reply by qerijo** [reply]
we don't have access to it due to the way The Smart Form component is designed unless we pass the methods inside it which looks kinda meh.

```
const methods = useForm()
<BasicSmartForm methods={methods} ...><Input name="n"></BasicSmartForm >
<Button onClick="methods.handleSubmit(cb)>Submit</Button>
```
That's not an optimal way to work with the form is that?

**Reply by bluebill1049** [reply]
not within the scope, unless you move it up.

**Reply by qerijo** [reply]
you mean wrapping the form component into <FormProvider {...methods} /> and extracting those in the component itself from the context instead of passing methods directly?
